README.md

```diff
@@ -172,3 +173,40 @@ We satisfy the SAP by:
 ```
   use_case = MyBankingApplication::UseCases::RetailCustomerMakesADeposit.new(input_port)
 ```
+
+## Why are there Monads in my Ruby code?
+
+We make use of the [Dry-Rb](https://dry-rb.org/) collection of Gems to
+provide better control flow instead of relying on `raise` and `rescue`.
+Specifically, we use:
+
+- https://dry-rb.org/gems/dry-matcher/result-matcher/
+- https://dry-rb.org/gems/dry-monads/1.0/result/
+
+### Things to consider for idiomatic FP code
+
+#### Multiple `bind` operations
+
+When you want to bind or chain multiple method calls using the
+previous return value, consider using [Do
+Notation](https://dry-rb.org/gems/dry-monads/1.0/do-notation/)
+
+This is inspired by the Haskell do-notation which lets you go from
+writing this:
+
+```haskell
+action1
+  >>=
+    (\ x1 -> action2
+       >>=
+         (\ x2 -> mk_action3 x1 x2 ))
+```
+
+to this:
+
+```haskell
+do
+  x1 <- action1
+  x2 <- action2
+  mk_action3 x1 x2
+```
```

Comment:
This is great, but I think if we consistently make addendums like this without a broader category/index it's going to end up not being as useful. I'm wondering if we should split the README into sections with Philosophy, Structure, Conventions, etc

Reply:
Valid feedback and I agree - I will whip it into shape :+1: I'll even add a "table of contents" :scream: :laughing: